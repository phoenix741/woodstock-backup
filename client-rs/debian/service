[Unit]
Description=Woodstock Client Daemon

[Service]
ExecStart=/usr/bin/ws_client_daemon

[Install]
WantedBy=network.target