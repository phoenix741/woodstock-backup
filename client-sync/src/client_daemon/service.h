#pragma once

#include <QtService>

#include "woodstockclient.h"

class ClientService : public QtService::Service
{
    Q_OBJECT
public:
    explicit ClientService(int &argc, char **argv);

protected:
    virtual ClientService::CommandResult onStart() override;
    virtual ClientService::CommandResult onStop(int &exitCode) override;

private:
    std::unique_ptr<WoodstockClient> m_client;
};
