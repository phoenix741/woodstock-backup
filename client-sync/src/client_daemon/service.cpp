#include "service.h"

ClientService::ClientService(int &argc, char **argv) : Service{argc, argv}, m_client(new WoodstockClient())
{
}

ClientService::CommandResult ClientService::onStart()
{
    qDebug() << "Service was started";

    m_client->listen();

    qDebug() << "Service is listening";
    return ClientService::CommandResult::Completed; //service is now assumed started
}

ClientService::CommandResult ClientService::onStop(int &exitCode)
{
    qDebug() << "Stop received";
    
    exitCode = 0;

    m_client->stop();

    qDebug() << "Stopped";
    return ClientService::CommandResult::Completed;
}
