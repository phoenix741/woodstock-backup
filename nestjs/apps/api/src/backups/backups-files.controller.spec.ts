import { Test, TestingModule } from '@nestjs/testing';
import * as archiver from 'archiver';
import * as httpMocks from 'node-mocks-http';
import { BackupsFilesController } from './backups-files.controller';
import { BackupsFilesService } from './backups-files.service';

jest.mock('archiver');

const mockedArchiver = archiver as jest.Mocked<typeof archiver>;

describe('Backups File Controller', () => {
  let controller: BackupsFilesController;

  const backupsFilesService = {
    listShare: (name: string, number: number) => [
      {
        name,
        dev: number,
      },
    ],
    list: (name: string, number: number, share: string, path: string) => [
      { name, dev: number },
      { name: share },
      { name: path },
    ],
    createArchive(archive: archiver.Archiver, name: string, number: number, share: string, path: string) {
      archive.append('test', { name: 'test' });
    },
  };

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [{ provide: BackupsFilesService, useValue: backupsFilesService }],
      controllers: [BackupsFilesController],
    }).compile();

    controller = module.get<BackupsFilesController>(BackupsFilesController);
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
  });

  it('should list available share', async () => {
    expect(await controller.share('name', 10)).toEqual([{ name: 'name', dev: 10 }]);
  });

  it('should list the available share of a path', async () => {
    expect(await controller.list('name', 10, 'share', 'path')).toEqual([
      { name: 'name', dev: 10 },
      { name: 'share' },
      { name: 'path' },
    ]);
  });

  it('should download zip file', async () => {
    // GIVEN
    const res = httpMocks.createResponse();
    const mockedArchiverInstance = {
      pipe: jest.fn(),
      append: jest.fn(),
      finalize: jest.fn(),
    } as unknown as archiver.Archiver;

    mockedArchiver.create.mockReturnValue(mockedArchiverInstance);

    // WHEN
    await controller.download('name', 10, 'share', 'path', res, 'application/zip');

    // THEN
    expect(mockedArchiver.create).toHaveBeenCalledWith('zip');
    expect(res).toMatchSnapshot('res');
    expect(mockedArchiverInstance.pipe).toHaveBeenCalledWith(res);
    expect(mockedArchiverInstance.append).toMatchSnapshot('append');
    expect(mockedArchiverInstance.finalize).toHaveBeenCalledWith();
  });

  it('should download zip file', async () => {
    // GIVEN
    const res = httpMocks.createResponse();
    const mockedArchiverInstance = {
      pipe: jest.fn(),
      append: jest.fn(),
      finalize: jest.fn(),
    } as unknown as archiver.Archiver;

    mockedArchiver.create.mockReturnValue(mockedArchiverInstance);

    // WHEN
    await controller.download('name', 10, 'share', 'path', res, 'application/x-tar');

    // THEN
    expect(mockedArchiver.create).toHaveBeenCalledWith('tar');
    expect(res).toMatchSnapshot('res');
    expect(mockedArchiverInstance.pipe).toHaveBeenCalledWith(res);
    expect(mockedArchiverInstance.append).toMatchSnapshot('append');
    expect(mockedArchiverInstance.finalize).toHaveBeenCalledWith();
  });
});
