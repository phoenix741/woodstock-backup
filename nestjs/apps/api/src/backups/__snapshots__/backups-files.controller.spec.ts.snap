// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backups File Controller should download zip file: append 1`] = `
[MockFunction] {
  "calls": [
    [
      "test",
      {
        "name": "test",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Backups File Controller should download zip file: append 2`] = `
[MockFunction] {
  "calls": [
    [
      "test",
      {
        "name": "test",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Backups File Controller should download zip file: res 1`] = `
EventEmitter {
  "_getBuffer": [Function],
  "_getChunks": [Function],
  "_getData": [Function],
  "_getHeaders": [Function],
  "_getJSONData": [Function],
  "_getLocals": [Function],
  "_getRedirectUrl": [Function],
  "_getRenderData": [Function],
  "_getRenderView": [Function],
  "_getStatusCode": [Function],
  "_getStatusMessage": [Function],
  "_headers": {
    "content-disposition": "attachment; filename="share_path.zip"",
    "content-type": "application/zip",
  },
  "_isDataLengthValid": [Function],
  "_isEndCalled": [Function],
  "_isJSON": [Function],
  "_isUTF8": [Function],
  "append": [Function],
  "appendHeader": [Function],
  "attachment": [Function],
  "clearCookie": [Function],
  "contentType": [Function],
  "cookie": [Function],
  "cookies": {},
  "destroy": [Function],
  "destroySoon": [Function],
  "end": [Function],
  "finished": false,
  "format": [Function],
  "get": [Function],
  "getEncoding": [Function],
  "getHeader": [Function],
  "getHeaderNames": [Function],
  "getHeaders": [Function],
  "hasHeader": [Function],
  "header": [Function],
  "headersSent": false,
  "json": [Function],
  "jsonp": [Function],
  "locals": {},
  "location": [Function],
  "redirect": [Function],
  "removeHeader": [Function],
  "render": [Function],
  "send": [Function],
  "sendStatus": [Function],
  "set": [Function],
  "setEncoding": [Function],
  "setHeader": [Function],
  "status": [Function],
  "statusCode": 200,
  "statusMessage": "OK",
  "type": [Function],
  "vary": [Function],
  "writableEnded": false,
  "writableFinished": false,
  "write": [Function],
  "writeHead": [Function],
}
`;

exports[`Backups File Controller should download zip file: res 2`] = `
EventEmitter {
  "_getBuffer": [Function],
  "_getChunks": [Function],
  "_getData": [Function],
  "_getHeaders": [Function],
  "_getJSONData": [Function],
  "_getLocals": [Function],
  "_getRedirectUrl": [Function],
  "_getRenderData": [Function],
  "_getRenderView": [Function],
  "_getStatusCode": [Function],
  "_getStatusMessage": [Function],
  "_headers": {
    "content-disposition": "attachment; filename="share_path.zip"",
    "content-type": "application/zip",
  },
  "_isDataLengthValid": [Function],
  "_isEndCalled": [Function],
  "_isJSON": [Function],
  "_isUTF8": [Function],
  "append": [Function],
  "appendHeader": [Function],
  "attachment": [Function],
  "clearCookie": [Function],
  "contentType": [Function],
  "cookie": [Function],
  "cookies": {},
  "destroy": [Function],
  "destroySoon": [Function],
  "end": [Function],
  "finished": false,
  "format": [Function],
  "get": [Function],
  "getEncoding": [Function],
  "getHeader": [Function],
  "getHeaderNames": [Function],
  "getHeaders": [Function],
  "hasHeader": [Function],
  "header": [Function],
  "headersSent": false,
  "json": [Function],
  "jsonp": [Function],
  "locals": {},
  "location": [Function],
  "redirect": [Function],
  "removeHeader": [Function],
  "render": [Function],
  "send": [Function],
  "sendStatus": [Function],
  "set": [Function],
  "setEncoding": [Function],
  "setHeader": [Function],
  "status": [Function],
  "statusCode": 200,
  "statusMessage": "OK",
  "type": [Function],
  "vary": [Function],
  "writableEnded": false,
  "writableFinished": false,
  "write": [Function],
  "writeHead": [Function],
}
`;
